packages/sign_in_with_apple/android/src/main/kotlin/com/aboutyou/dart_packages/sign_in_with_apple/SignInWithApplePlugin.kt

```diff
@@ -145,32 +168,62 @@ public class SignInWithApplePlugin: FlutterPlugin, MethodCallHandler, ActivityAw
  *
  * DO NOT rename this or it's package name as it's configured in the consumer's `AndroidManifest.xml`
  */
-public class SignInWithAppleCallback: Activity {
-  constructor() : super()
+public class SignInWithAppleCallback : Activity() {
+  private fun validateNonce(uri: Uri, expectedNonce: String?): Boolean {
+    val idToken = uri.getQueryParameter("id_token")
+    if (idToken == null) {
+      Log.e(TAG, "Missing id_token query parameter in signinwithapple callback")
+      return false
+    }
+
+    try {
+      val jwt = JWT(idToken)
+
+      if (jwt.getClaim("nonce_supported").asBoolean() == true) {
+        if (jwt.getClaim("nonce").asString() != expectedNonce) {
+          Log.e(TAG, "Expected that the JWT nonce matches the initially provided one, but they differed")
+          return false
+        }
+      }
```

Comment:
```suggestion
      } else if (expectedNonce != null) {
          Log.e(TAG, "Expected a nonce on the token, but none was included")
          return false
      }
```

I think this case was not handled yet. If the token didn't have a nonce, it wouldn't run into the detailed check in the first place.